winter_messaging_transactional/consumer/inbox/inbox_message_dao.py

```diff
@@ -37,10 +39,10 @@ def upsert(self, event: InboxMessage) -> InboxResult:
             inbox_message_table.c.counter,
             inbox_message_table.c.processed_at,
         )
-        with self._engine.connect() as connection:
-            result = connection.execute(statement)
-            inserted_record = result.fetchone()
-            return InboxResult(*inserted_record)
+        result = self._session.execute(statement)
+        self._session.flush()
```

Comment:
i'm not sure that this call of flush method make sense here. it would be necessary if we would work in orm mode with entities instead of it we execute query directly on db side and fetch result as tupes